src/log/Log.cc

```diff
@@ -227,25 +227,23 @@ void Log::_flush(EntryQueue *t, EntryQueue *requeue, bool crash)
       buflen += snprintf(buf + buflen, sizeof(buf)-buflen, " %lx %2d ",
 			(unsigned long)e->m_thread, e->m_prio);
 
-      // FIXME: this is slow
-      string s = e->get_str();
-
-      if (do_fd) {
-	int r = safe_write(m_fd, buf, buflen);
-	if (r >= 0)
-	  r = safe_write(m_fd, s.data(), s.size());
-	if (r >= 0)
-	  r = write(m_fd, "\n", 1);
-	if (r < 0)
-	  cerr << "problem writing to " << m_log_file << ": " << cpp_strerror(r) << std::endl;
-      }
+      buflen += e->snprintf(buf + buflen, sizeof(buf) - buflen - 1 );
```

Comment:
I have tried using writev but it is actually slower.
Below using writev vs using memcpy to make 1 buffer.
size=  6 IOVS=3 time=107768usec
size=  6 WRITE  time=84712usec
size= 12 IOVS=3 time=131376usec
size= 12 WRITE  time=120907usec
size= 24 IOVS=3 time=183488usec
size= 24 WRITE  time=172170usec
size= 48 IOVS=3 time=271790usec
size= 48 WRITE  time=271460usec
size= 96 IOVS=3 time=674332usec
size= 96 WRITE  time=424097usec
size=192 IOVS=3 time=769107usec
size=192 WRITE  time=784972usec
size=384 IOVS=3 time=2253296usec
size=384 WRITE  time=1509636usec
size=768 IOVS=3 time=2977487usec
size=768 WRITE  time=2974786usec